### PR TITLE
feat(project-creation): solidify creation logic

### PR DESCRIPTION
**Pull request by sentaur-athena**
There are multiple places in the codebase that decide if we want to show "create project" button to the user. This commit solidifies the logic for 2 reasons:
1. We don't want these paths to have different logic. Eligibility has to stay the same no matter what product is calling it.
2. We're going to experiment with showing the button to more people and this helps keep experimentation code in one place too.

### CODE REVIEW

File: static/app/components/projects/utils.spec.jsx
Comment (sentaur-athena) [inline static/app/components/projects/utils.spec.jsx:13]: @leedongwei is there a better way to create an org object for normal non-admin member? I just mocked the value of access here.

Comment (leedongwei) [reply static/app/components/projects/utils.spec.jsx:13]: I don't think there's a better way. Mocking `access` is okay.

File: static/app/components/projects/utils.tsx
Comment (leedongwei) [inline static/app/components/projects/utils.tsx:6]: We've not used `Viewer` as a noun to describe the person interacting with the app. 

@evanpurkhiser any opinions on naming?

**Review by leedongwei — APPROVED** [review]
LGTM. 

Minor possible nit on naming convention, let's wait for a response before merging.

File: static/app/components/projects/utils.tsx
Comment (nhsiehgit) [reply static/app/components/projects/utils.tsx:6]: is `User` too confusing?

as far as actor nouns go
- Customer - the org entity
- User - the individual user entity

Maybe we solidify this in our [notion glossary](https://www.notion.so/sentry/Glossary-bdc7209036d14e2799e3842b7bd590ef) 😂

Comment (evanpurkhiser) [reply static/app/components/projects/utils.tsx:6]: I would just call it `canCreateProject`, this is what we typically call stuff like this. The name of the parameter helps distinguish.

**Pull request merged**
